src/VisualStudio/Core/Def/LanguageServer/Handler/CustomMessage/CustomMessageDocumentHandler.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CustomMessageHandler;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.CodeAnalysis.Remote;
+using Roslyn.LanguageServer.Protocol;
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler.CustomMessage;
+
+[ExportCSharpVisualBasicStatelessLspService(typeof(CustomMessageDocumentHandler)), Shared]
+[Method(MethodName)]
+[method: ImportingConstructor]
+[method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+internal class CustomMessageDocumentHandler()
```

Comment:
I may have asked this before and just forgot the answer - can we move the LSP implementation of this into MS.CA.LanguageServer.Protocol?

That way we can share the impl among all our lsp servers.

Reply:
Done. I will test shortly and reach out if this is breaking anything.